services/sharedchannel/service.go

```diff
@@ -177,3 +178,16 @@ func (scs *Service) pause() {
 
 	scs.server.GetLogger().Debug("Shared Channel Service inactive")
 }
+
+// Makes the remote channel to be read only, only for guests.
```

Comment:
Did you mean remote users here? Guests have a different meaning.

Reply:
Took the reference for the definition of read-only-ness that we have in the channel permissions, we have a different layer for Guest users and Channel members, and I've implemented the read-only aspect to apply to Guests. Will amend that since Doug clarified that we want the channel to be an announcements channel so only admins can Post or React to the channel.

Reply:
amended in https://github.com/mattermost/mattermost-server/pull/16803/commits/749ec630122e22d8f49487d7c88457cbc7ba70f4

Reply:
Ah so it's actually for guests. Thanks.

Reply:
it was yes, but after the clarification, we want the readonly-ness to apply to all apart from admins.